src/main/java/org/maxgamer/quickshop/database/DatabaseManager.java

```diff
@@ -58,84 +59,150 @@
      * Queued database manager. Use queue to solve run SQL make server lagg issue.
      *
      * @param plugin plugin main class
-     * @param db     database
+     * @param dbCore database core
      */
-    public DatabaseManager(@NotNull QuickShop plugin, @NotNull Database db) {
+    public DatabaseManager(@NotNull QuickShop plugin, @NotNull AbstractDatabaseCore dbCore) throws ConnectionException {
         this.plugin = plugin;
         this.warningSender = new WarningSender(plugin, 600000);
-        this.database = db;
+        if (!dbCore.getConnection().isValid()) {
+            throw new DatabaseManager.ConnectionException("The database does not appear to be valid!");
+        }
+        this.database = dbCore;
         this.useQueue = plugin.getConfig().getBoolean("database.queue");
 
-        if (!useQueue) {
-            return;
+        if (useQueue) {
+            try {
+                task =
+                        new BukkitRunnable() {
+                            @Override
+                            public void run() {
+                                plugin.getDatabaseManager().runTask();
+                            }
+                        }.runTaskTimerAsynchronously(plugin, 1, plugin.getConfig().getLong("database.queue-commit-interval") * 20);
+            } catch (IllegalPluginAccessException e) {
+                Util.debugLog("Plugin is disabled but trying create database task, move to Main Thread...");
+                plugin.getDatabaseManager().runTask();
+            }
         }
-        try {
-            task =
-                    new BukkitRunnable() {
-                        @Override
-                        public void run() {
-                            plugin.getDatabaseManager().runTask();
-                        }
-                    }.runTaskTimerAsynchronously(plugin, 1, plugin.getConfig().getLong("database.queue-commit-interval") * 20);
-        } catch (IllegalPluginAccessException e) {
-            Util.debugLog("Plugin is disabled but trying create database task, move to Main Thread...");
-            plugin.getDatabaseManager().runTask();
+    }
+
+    /**
+     * Returns true if the table exists
+     *
+     * @param table The table to check for
+     * @return True if the table is found
+     * @throws SQLException Throw exception when failed execute somethins on SQL
+     */
+    boolean hasTable(@NotNull String table) throws SQLException {
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/Ghost-chu/QuickShop-Reremake/pullRequest?prid=6383074)